### PR TITLE
React example

### PR DESCRIPTION
**Pull request by VirgilChindris**
This is an example of adding and removing plots, using [React](https://en.wikipedia.org/wiki/React_(JavaScript_library)).

### CODE REVIEW

File: examples/reactGraph.html [hunk 1]
```diff
@@ -0,0 +1,245 @@
...
+
+            ni-scale-legend {
+                position: absolute;
+                left: 799px;
+                top: 328px;
+                height: 70px;
+            }
+
+            ni-cursor-legend {
+                /*position: absolute;*/
```
Comment (ralucaportase) [inline examples/reactGraph.html]: Please remove commented line

File: index.html [hunk 2]
```diff
@@ -38,6 +38,7 @@ <h2>Webcharts examples</h2>
                     <tr><td style="padding-left: 1em"><a href="examples/intensity.html">Intensity</a></td></tr>
                     <tr><td style="padding-left: 1em"><a href="examples/bouncingcog.html">XY Graph</a></td></tr>
                     <tr><td style="padding-left: 1em"><a href="examples/multisim_parity.html">Multisim parity</a></td></tr>
+                    <tr><td style="padding-left: 1em"><a href="examples/reactGraph.html">Graph using React</a></td></tr>
```
Comment (cristiingineru) [inline index.html]: Actually the graphs are not using react. The example is using a graph with react.

File: examples/reactGraph.html [hunk 3]
```diff
@@ -0,0 +1,245 @@
...
+                width: 400px;
+                left: 0px;
+                top: 399px;
+            }
+        </style>
+    </head>
+
+    <body>
+        <h1>Webcharts Graph</h1>
```
Comment (cristiingineru) [inline examples/reactGraph.html]: Make sure the title somehow match the h1 and the name of the example specified in index.html

File: examples/reactGraph.html [hunk 4]
```diff
@@ -0,0 +1,245 @@
...
+                                    <ni-cartesian-axis show show-ticks show-minor-ticks show-label label="Amplitude" axis-position="left" auto-scale="exact" grid-lines></ni-cartesian-axis>
+                                    <PlotList plotCodes={this.props.plots} />
+                                </ni-cartesian-graph>
+
+                                <ni-scale-legend graph-ref="28"></ni-scale-legend>
+                                <ni-plot-legend graph-ref="28"></ni-plot-legend>
+                                <ni-graph-tools graph-ref="28"></ni-graph-tools>
+                                <ni-light-plot-legend show graph-ref="28" position="nw"></ni-light-plot-legend>
+                                <br/>
+                                <ni-cursor-legend graph-ref="28"></ni-cursor-legend>
```
Comment (cristiingineru) [inline examples/reactGraph.html]: React is known to be a good choice for a large app, in this case almost the entire app is under a single React component.

You should try moving at least on of the legends (ni-cursor-legend maybe) in its own React component.

File: examples/reactGraph.html [hunk 5]
```diff
@@ -0,0 +1,245 @@
...
+                        color += letters[Math.floor(Math.random() * 16)];
+                    }
+                    return color;
+                }
+
+                genPlotContent = (id) => {
+                    return {id: id, plotLabel: 'Plot_' + id, plotColor: this.getRandomColor(), data: this.genPlotDataValues()}
+                }
+
+                genDummyPlotContent = () => {
```
Comment (ralucaportase) [inline examples/reactGraph.html]: Could you please add an explanation for why is this needed. Maybe even rename it or something to make it more intuitive :)

File: examples/reactGraph.html (hunk 1 above)
Comment (VirgilChindris) [reply examples/reactGraph.html]: Done

File: examples/reactGraph.html (hunk 5 above)
Comment (VirgilChindris) [reply examples/reactGraph.html]: Done

File: examples/reactGraph.html (hunk 3 above)
Comment (VirgilChindris) [reply examples/reactGraph.html]: Done

File: index.html (hunk 2 above)
Comment (VirgilChindris) [reply index.html]: Done
